hikari/impl/entity_factory.py

```diff
@@ -1777,6 +1821,27 @@ def _deserialize_interaction_command_option(
             options=suboptions,
         )
 
+    def _deserialize_autocomplete_interaction_option(
+        self,
+        payload: data_binding.JSONObject,
+    ) -> command_interactions.AutocompleteInteractionOption:
+        suboptions: typing.Optional[typing.List[command_interactions.CommandInteractionOption]] = None
+        if raw_suboptions := payload.get("options"):
+            suboptions = [self._deserialize_interaction_command_option(suboption) for suboption in raw_suboptions]
+
+        option_type = commands.OptionType(payload["type"])
+        value = payload.get("value")
+        if modifier := _interaction_option_type_mapping.get(option_type):
+            value = modifier(value)
```

Comment:
This is missing a test case for if the option value is cast by a modifier

Reply:
Also as per https://github.com/discord/discord-api-docs/issues/4338, this should check if "focused" is True before casting as it won't be guaranteed to be valid if its the focused option (with a test case for this as well). This behaviour might also be worth documenting on a lib level if it isn't already

Reply:
Still missing the test coverage here

Reply:
Surely this should take care of that right? 
https://github.com/thesadru/hikari/blob/8142813d459229941d96199c4b7d6ad96419dba6/tests/hikari/impl/test_entity_factory.py#L3177